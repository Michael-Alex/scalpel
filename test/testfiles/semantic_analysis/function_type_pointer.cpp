typedef int (*const fun_ptr_t)();

int (*const fun_ptr)();
fun_ptr_t fun_ptr2;

void f(void (*fun_ptr)(int))
{
}

