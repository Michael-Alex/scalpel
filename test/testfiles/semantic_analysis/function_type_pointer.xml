<namespace id="n0">
 <typedef name="fun_ptr_t">
  <type>
   <cv_qualified_type const="true">
    <pointer>
     <function_type>
      <return_type>
       <fundamental_type type="int"/>
      </return_type>
      <parameter_types>
      </parameter_types>
     </function_type>
    </pointer>
   </cv_qualified_type>
  </type>
 </typedef>
 <simple_function name="f">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter name="fun_ptr">
    <type>
     <pointer>
      <function_type>
       <return_type>
        <fundamental_type type="void"/>
       </return_type>
       <parameter_types>
        <fundamental_type type="int"/>
       </parameter_types>
      </function_type>
     </pointer>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function name="g" defined="false">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <pointer>
      <function_type>
       <return_type>
        <fundamental_type type="int"/>
       </return_type>
       <parameter_types>
        <fundamental_type type="double"/>
       </parameter_types>
      </function_type>
     </pointer>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <variable name="fun_ptr">
  <type>
   <cv_qualified_type const="true">
    <pointer>
     <function_type>
      <return_type>
       <fundamental_type type="int"/>
      </return_type>
      <parameter_types>
      </parameter_types>
     </function_type>
    </pointer>
   </cv_qualified_type>
  </type>
 </variable>
 <variable name="fun_ptr2">
  <type>
   <cv_qualified_type const="true">
    <pointer>
     <function_type>
      <return_type>
       <fundamental_type type="int"/>
      </return_type>
      <parameter_types>
      </parameter_types>
     </function_type>
    </pointer>
   </cv_qualified_type>
  </type>
 </variable>
</namespace>
