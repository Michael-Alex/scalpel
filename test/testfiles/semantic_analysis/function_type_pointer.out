<namespace id="0">
 <typedef id="0" name="fun_ptr_t">
  <type>
   <cv_qualified_type const="true">
    <pointer>
     <function_type>
      <return_type>
       <fundamental_type type="int"/>
      </return_type>
      <parameter_types>
      </parameter_types>
     </function_type>
    </pointer>
   </cv_qualified_type>
  </type>
 </typedef>
 <simple_function id="0" name="f">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter name="fun_ptr">
    <type>
     <pointer>
      <function_type>
       <return_type>
        <fundamental_type type="void"/>
       </return_type>
       <parameter_types>
        <fundamental_type type="int"/>
       </parameter_types>
      </function_type>
     </pointer>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <simple_function id="1" name="g" defined="false">
  <return_type>
   <fundamental_type type="void"/>
  </return_type>
  <parameters>
   <parameter>
    <type>
     <pointer>
      <function_type>
       <return_type>
        <fundamental_type type="int"/>
       </return_type>
       <parameter_types>
        <fundamental_type type="double"/>
       </parameter_types>
      </function_type>
     </pointer>
    </type>
   </parameter>
  </parameters>
 </simple_function>
 <variable id="0" name="fun_ptr">
  <type>
   <cv_qualified_type const="true">
    <pointer>
     <function_type>
      <return_type>
       <fundamental_type type="int"/>
      </return_type>
      <parameter_types>
      </parameter_types>
     </function_type>
    </pointer>
   </cv_qualified_type>
  </type>
 </variable>
 <variable id="1" name="fun_ptr2">
  <type>
   <cv_qualified_type const="true">
    <pointer>
     <function_type>
      <return_type>
       <fundamental_type type="int"/>
      </return_type>
      <parameter_types>
      </parameter_types>
     </function_type>
    </pointer>
   </cv_qualified_type>
  </type>
 </variable>
</namespace>
